app/src/main/java/com/example/hris/ui/viewmodels/AddTimeLogsViewModel.kt

```diff
@@ -0,0 +1,37 @@
+package com.example.hris.ui.viewmodels
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.example.hris.model.AddTimeLogs
+import com.example.hris.network.HrisApi
+import com.example.hris.ui.DialogState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class AddTimeLogsViewModel @Inject constructor(
+    application: Application
+) : AndroidViewModel(application) {
+
+    val loadingDialogState = MutableLiveData<DialogState>()
+    val callValue = MutableLiveData<AddTimeLogs>()
+
+    fun addTimeLogs(userId: String, type: String) {
+        viewModelScope.launch {
+            loadingDialogState.value = DialogState.SHOW
+            callApi(userId, type)
+        }
+    }
+
+    private suspend fun callApi(userId: String, type: String) {
+        val call = HrisApi.retrofitService.addTimeLogs(
+            userId,
+            type
+        )
```

Comment:
This is a bad pattern. The purpose of the Repository layer is to provide the data. Whether those data came from a local database or from a remote API should not be a concern of the ViewModel. Hence, the HrisApi should only be called inside the Repository then the Repository should be injected into the ViewModels